### ISSUE TITLE
False positive of uncessary-comprehension when repacking tuple members with extra element

### ISSUE THREAD

**Issue by vinzd** [PyCQA/pylint#3148]
### Steps to reproduce
```
my_list = [(1, 2), (3, 4)]
new_list = [(low, end, None) for low, end in my_list]
```

Please notice that the issue is not reproduced with the following example:
```
my_list = [(1, 2), (3, 4)]
my_var = None
new_list = [(low, end, my_var) for low, end in my_list]
```

### Current behavior
```
************* Module example
example.py:2: [R1721(unnecessary-comprehension), ] Unnecessary use of a comprehension
```

### Expected behavior
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### pylint --version output
```
pylint 2.4.2
astroid 2.3.1
Python 3.7.4 (default, Jul 16 2019, 07:12:58) 
[GCC 9.1.0]
```

**Comment by RemiCardona** [comment]
```python
[(x, y, 1) for x, y in iterable]
```

Adding the above snippet to `tests/functional/u/unnecessary_comprehension.py` reproduces the issue
